src/content/docs/react/components.mdx

```diff
@@ -1,3 +1,162 @@
 ---
 title: Components in React
 ---
+
+## What are Components?
+
+Components are the building blocks of any React application. They encapsulate logic and UI, allowing developers to create reusable, modular pieces of an application. Each component can maintain its own state, receive data via props, and manage its rendering, making it easy to compose complex UIs from simple building blocks. By breaking down the UI into smaller components, React promotes reusability and better maintainability.
+
+## Functional Components vs. Class Components
+
+React supports two types of components: **functional components** and **class components**.
+
+### Functional Components
+
+Functional components are JavaScript functions that return JSX. They are simpler and easier to read, especially when using React Hooks for state management and side effects.
+
+**Example:**
+
+```javascript
+function Greeting(props) {
+  return <h1>Hello, {props.name}!</h1>;
+}
+```
+
+### Class Components
+
+Class components are ES6 classes that extend `React.Component`. They can hold and manage their own state and lifecycle methods.
+
+**Example:**
+
+```javascript
+import React from 'react';
+
```

Comment:
Please use the `src\content\docs\react\codes\componentsCode` folder to dump all code-related files. Please use `mdx` not just `md`. Use Astro Star Light components such as tab, code etc to present the content visually.